modules/auxiliary/scanner/http/docker_enum.rb

```diff
@@ -0,0 +1,49 @@
+##
+# This module requires Metasploit: https://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+class MetasploitModule < Msf::Auxiliary
+  include Msf::Exploit::Remote::HttpClient
+  include Msf::Auxiliary::Scanner
+  include Msf::Auxiliary::Report
+
+  def initialize
+    super(
+      'Name'        => 'Docker Server Version Scanner',
+      'Description' => 'This module attempts identify the version of the Docker Server running.',
+      'Author'     => [ 'Agora-Security' ],
+      'License'   => MSF_LICENSE
+    )
+    register_options(
+      [
+        Opt::RPORT(2375),
+      ])
+    register_autofilter_ports([ 2375 ])
+    deregister_options('RHOST')
+  end
+
+  def run_host(ip)
+    res = send_request_cgi({
+      'uri'    => normalize_uri(datastore['URI'], "/version"),
```

Comment:
Did you mean `TARGETURI`? It doesn't look like `URI` is coming from anywhere? `TARGETURI` should be registered, and you should use `target_uri.path`.

Reply:
Fixed in 5a89642ddd.

I also did some "major" simplification of the module.